Source/WebCore/platform/graphics/FloatQuad.cpp

```diff
@@ -109,6 +109,21 @@ bool FloatQuad::containsQuad(const FloatQuad& other) const
     return containsPoint(other.p1()) && containsPoint(other.p2()) && containsPoint(other.p3()) && containsPoint(other.p4());
 }
 
+bool FloatQuad::intersectsQuad(const FloatQuad& other) const
+{
+    for (const auto& p : { other.p1(), other.p2(), other.p3(), other.p4() }) {
+        if (containsPoint(p))
+            return true;
+    }
```

Comment:
Hm....I don't think this takes care of some corner cases (such as this case):

```
░░░░░░█████████████░░░░
░░░░░░█░░░░░░░░░░░█░░░░
░░░░░░█░░░░░░░░░░░█░░░░
░│││││█│││││││││││█│││░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│░░░░█░░░░░░░░░░░█░░│░
░│││││█│││││││││││█│││░
░░░░░░█░░░░░░░░░░░█░░░░
░░░░░░█░░░░░░░░░░░█░░░░
░░░░░░█░░░░░░░░░░░█░░░░
░░░░░░█████████████░░░░
```

(It might also be good to write a quick unit test case for this in `FloatQuadTests.cpp`)